src/Application.cpp

```diff
@@ -373,6 +373,67 @@ namespace weatherserver {
         location.setIsAddingWeatherToAddressSpace(false);
     }
 
+    /*
+        Checking if locations number in the server model matches with actual json parse. Adjusting if needed.
+     */
+    static void validateLocationsNumberInTheModel(UA_Server& server, CountryData& country, int actualParseSize) {
+
+        //crafting NodeID for our locations number attribute inside the information model
+        std::string locationsNumberAttributeString = std::string(CountryData::COUNTRIES_FOLDER_NODE_ID) + "."
+            + country.getCode() + "." + std::string(CountryData::BROWSE_LOCATIONS_NUMBER);
+        std::cout << "Checking locations number for the following ID: " << locationsNumberAttributeString << std::endl;
+        UA_NodeId locationsNumberAttributeNodeId = UA_NODEID_STRING(WebService::OPC_NS_INDEX,
+            const_cast<char*>(locationsNumberAttributeString.c_str()));
+
+        UA_Variant valueInModel;
+        UA_Variant_init(&valueInModel);
+        UA_StatusCode retvalRead = UA_Server_readValue(&server, locationsNumberAttributeNodeId, &valueInModel);
+
+        if (retvalRead == UA_STATUSCODE_GOOD && UA_Variant_hasScalarType(&valueInModel, &UA_TYPES[UA_TYPES_UINT32])) {
+
+            UA_UInt32 valueInModelUInt = *(UA_UInt32*)valueInModel.data;
+            std::cout << "Value in the information model: " << valueInModelUInt << std::endl;
+            UA_UInt32 valueJsonUInt = actualParseSize;
+            std::cout << "Value after parsing json file: " << valueJsonUInt << std::endl;
+
+            if (valueInModelUInt != valueJsonUInt) {
+                std::cout << "Number of locations in the information model doesn't match one after parsing. Trying to adjust..." << std::endl;
+                UA_Variant_setScalarCopy(&valueInModel, &valueJsonUInt, &UA_TYPES[UA_TYPES_UINT32]);
+                UA_StatusCode retvalWrite = UA_Server_writeValue(&server, locationsNumberAttributeNodeId, valueInModel);
+
+                if (retvalWrite == UA_STATUSCODE_GOOD) {
+                    //Re-read the value from the model to make sure we got it right.
```

Comment:
Do we really need to read and check again if `UA_Server_writeValue` returned `UA_STATUSCODE_GOOD`?

Reply:
Probably not. Didn't like that part anyway. Added just to make sure works as intended, but now seems unnecessary. Just going to set class parameter there if code is good.